src/js/components/tabs.js

```diff
@@ -0,0 +1,220 @@
+    const focusList = nodeListToArray(this.tabs);
+    const nextTab = focusList.indexOf(currentTab) + 1;
+    const prevTab = focusList.indexOf(currentTab) - 1;
+
+    switch (event.keyCode) {
+      case keyCodes.right || keyCodes.down: {
+
+        !focusList[nextTab] ? focusList[0].focus() : focusList[nextTab].focus();
+
+        break;
+      }
+      case keyCodes.down: {
+
+        !focusList[nextTab] ? focusList[0].focus() : focusList[nextTab].focus();
+
+        break;
+      }
+      case keyCodes.left: {
+
+        !focusList[prevTab] ? focusList[focusList.length - 1].focus() : focusList[prevTab].focus();
+
+        break;
+      }
+      case keyCodes.up: {
+
+        !focusList[prevTab] ? focusList[focusList.length - 1].focus() : focusList[prevTab].focus();
+
+        break;
+      }
+      case keyCodes.end: {
+
+        focusList[focusList.length - 1].focus();
+
+        break;
+      }
+      case keyCodes.home: {
+
+        focusList[0].focus();
+
+        break;
+      }
+      default: {
+        break;
+      }
+    }
+  }
+
+  /**
+   * This function is used to trigger the 'tabActivated' custom event,
+   * deactivate any other tab, and activate a tab/panel pair. It removes the
+   * panel's hidden attribute, adds the .rvt-tab-open class, sets the tab's
+   * aria-selected attribute to true. It also allows developers to setup a
+   * custom callback function.
+   * @param {Function} callback 
+   */
+  activateTab(tab, callback) {
+    // Trigger tabActivated custom event. This event is used to control the process of switching between tabs.
+
+    const activationEvent = dispatchCustomEvent(
+      'tabActivated',
+      tab,
+      {
+        id: tab.dataset.tabPanel,
+      }
+    );
+
+    if (!activationEvent) return;
+
+    const trigger = this.element.querySelector(`[${this.tabAttribute}="${tab.dataset.tabPanel}"]`);
+
+    this.panels.forEach(panel => {
+      if (panel.getAttribute(this.panelAttribute) !== tab.dataset.tabPanel && !panel.hasAttribute('hidden')) {
+        this.deactivateTab(panel);
+      }
+    });
+
+    // Activate the appropriate tab/panel pair
+    tab.removeAttribute('hidden');
+    tab.classList.add('rvt-tab-open');
+    trigger.setAttribute('aria-selected', 'true');
+    trigger.removeAttribute('tabindex');
+
+    if (callback && typeof callback === 'function') {
+      callback();
+    }
+  }
+
+  /**
+   * This function is used to trigger the 'tabDeactivated' custom event,
+   * and deactivates a tab/panel pair. It adds the hidden attribute, removes
+   * the .rvt-tab-open class, and sets the tab's aria-selected attribute to
+   * false. It also allows developers to setup a custom callback function.
+   * @param {Function} callback 
+   */
+  deactivateTab(tab, callback) {
+    // Trigger tabDeactivated custom event. This event is used to control the process of switching between tabs.
+
+    const deactivationEvent = dispatchCustomEvent(
```

Comment:
For example, I don't feel like we need a `tabDeactivated` event in addition to a `tabActivated` event. If you look at where you're calling `this.deactivateTab` in the `activateTab` method there's really nothing functionally going on other than querying the DOM for the toggle element.

Also, the way the `activateTab` method is set up right now, the `tabDeactivated` Event will be fired for every tab except the one that get activated. So in the example `tab.njk` it gets fire three times in a row. 

I'd say for now let's just stick with `tabActivated` for the custom events.

Without this custom event logic I'd say we should definitely get rid of the this `deactivateTab` method.